src/util/redux.spec.js

```diff
@@ -0,0 +1,197 @@
+				const nextState = reducer(initialState, action);
+				assert.equal(nextState.foo.bar.value, 'baz');
+			});
+
+		});
+
+		describe('with rootPath', () => {
+
+			const { reducer: reducerWithRootPath } = getReduxPrimitives({
+				reducers,
+				initialState,
+				rootPath: ['root']
+			});
+
+			it('should correctly apply state change', () => {
+				const action = { type: 'root.foo.onChange', payload: 'bar' };
+				const nextState = reducerWithRootPath(initialState, action);
+				assert.equal(nextState.foo.value, 'bar');
+			});
+
+		});
+
+		describe('thunks', () => {
+			it('should not include thunk paths in reducer', () => {
+				const action = { type: 'root.foo.asyncOperation' };
+				const nextState = reducer(initialState, action);
+				assert.deepEqual(initialState, nextState);
+			});
+		});
+
+	});
+
+	describe('connectors', () => {
+		describe('selector/mapStateToProps', () => {
+
+			const selectors = {
+				foo: {
+					uppercase: ({ value }) => upperCase(value)
+				}
+			};
+
+			const {
+				connectors: [mapStateToProps]
+			} = getReduxPrimitives({
+				reducers,
+				initialState,
+				selectors
+			});
+
+			it('should apply selector', () => {
+				const viewState = mapStateToProps(initialState);
+				assert.equal(viewState.foo.uppercase, 'FOO');
+			});
+
+			describe('rootPath', () => {
+
+				const {
+					connectors: [mapStateToProps]
+				} = getReduxPrimitives({
+					reducers,
+					initialState,
+					selectors,
+					rootPath: ['root']
+				});
+
+				it('should apply selector', () => {
+					const viewState = mapStateToProps({ root: initialState });
+					assert.equal(viewState.foo.uppercase, 'FOO');
+				});
+
+			});
+
+			describe('rootSelector', () => {
+
+				const {
+					connectors: [mapStateToProps]
+				} = getReduxPrimitives({
+					reducers,
+					initialState,
+					selectors,
+					rootSelector: state => assign({}, state, {
+						computed: state.foo.uppercase + state.foo.value
+					})
+				});
+
+				it('should apply rootSelector', () => {
+					const viewState = mapStateToProps(initialState);
+					assert.equal(viewState.computed, 'FOOfoo');
+				});
+
+			});
+		});
+
+		describe('dispatchTree/mapDispatchToProps', () => {
+
+			const {
+				connectors
+			} = getReduxPrimitives({
+				reducers,
+				initialState
+			});
+
+			const mapDispatchToProps = connectors[1];
+			const mockDispatch = sinon.spy(action => isFunction(action) ? action(mockDispatch) : action);
+			const dispatchTree = mapDispatchToProps(mockDispatch);
+
+			beforeEach(() => mockDispatch.reset());
+
+			it('should dispatch the correct action', () => {
+				dispatchTree.foo.onChange('bar', 'baz');
+				const dispatchedAction = mockDispatch.getCall(0).args[0];
+				assert.deepEqual(dispatchedAction, {
+					type: 'foo.onChange',
+					payload: 'bar',
+					meta: ['baz']
+				});
+			});
+
+			describe('thunks', () => {
+
+				beforeEach(() => dispatchTree.foo.asyncOperation('qux'));
```

Comment:
i thought we were on a version of mocha where you can return promises and it'll handle completion and errors correctly